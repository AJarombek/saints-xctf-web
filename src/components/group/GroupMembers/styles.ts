/**
 * JSS styles for the GroupMembers component.
 * @author Andrew Jarombek
 * @since 1/10/2021
 */

import Colors from '../../../styles/colors';
import { FontMixins } from '../../../styles/mixins';

export default {
  container: {
    marginTop: 20,
    cursor: 'pointer'
  },
  member: {
    display: 'flex',
    alignItems: 'center',
    borderRadius: 4,
    marginBottom: 10,
    boxShadow: 'rgba(170, 170, 170, 0.2) 0px 1px 3px',
    backgroundColor: Colors.lightestBackground,

    '& > p:nth-child(1)': {
      ...FontMixins.robotoSlab(),
      marginLeft: 20
    },

    '& > p:nth-child(2)': {
      ...FontMixins.robotoBold(),
      margin: '0 20px 0 auto',
      fontSize: 14
    }
  }
};
