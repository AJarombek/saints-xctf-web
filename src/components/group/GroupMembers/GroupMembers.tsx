/**
 * Component for a tab on the group page which lists all the users who are members.
 * @author Andrew Jarombek
 * @since 1/10/2021
 */

import React from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import { MemberDetails } from '../../../redux/types';
import moment from 'moment';
import { useHistory } from 'react-router-dom';

const useStyles = createUseStyles(styles);

interface Props {
  members: MemberDetails[];
}

const GroupMembers: React.FunctionComponent<Props> = ({ members }) => {
  const history = useHistory();
  const classes = useStyles();

  return (
    <div className={classes.container}>
      {members.map((member: MemberDetails) => (
        <div
          key={member.username}
          className={classes.member}
          onClick={(): void => history.push(`/profile/${member.username}`)}
        >
          <p>
            {member.first} {member.last}
          </p>
          <p>Member Since: {moment(member.member_since).format('MMM Do, YYYY')}</p>
        </div>
      ))}
    </div>
  );
};

export default GroupMembers;
