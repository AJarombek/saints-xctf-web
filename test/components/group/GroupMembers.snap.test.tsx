/**
 * Snapshot test for the GroupMembers component.
 * @author Andrew Jarombek
 * @since 3/4/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import GroupMembers from '../../../src/components/group/GroupMembers';

describe('GroupMembers Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <GroupMembers
            members={[
              {
                username: 'andy',
                first: 'Andy',
                last: 'Jarombek',
                member_since: '2016-12-23',
                user: 'admin',
                status: 'accepted'
              }
            ]}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
