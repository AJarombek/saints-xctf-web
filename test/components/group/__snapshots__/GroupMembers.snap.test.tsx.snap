// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupMembers Snapshot Tests renders correctly 1`] = `
<div
  className="container-0-1-1"
>
  <div
    className="member-0-1-2"
    onClick={[Function]}
  >
    <p>
      Andy
       
      Jarombek
    </p>
    <p>
      Member Since: 
      Dec 23rd, 2016
    </p>
  </div>
</div>
`;
